File: Python/7py 1/disarium_number.py
"""
|--------------------------------------------------------------------------
| Disarium Number (Special Numbers Series #3)
|--------------------------------------------------------------------------
|
"""

def disarium_number(number):
   string = str(number)
   res = 0
   for i in range(len(string)):
      res += int(string[i]) ** (i+1)
   return "Disarium !!" if res == number else "Not !!"


def disarium_number(number):
   return "Disarium !!" if sum(int(n)**i for i, n in enumerate(str(number), 1)) == number else "Not !!"


def disarium_number(number):
   a=0
   for j,i in enumerate(str(number)):
      a+=int(i)**(j+1)
   return 'Disarium !!' if a==number else 'Not !!'


def disarium_number(num):
   total = sum(int(n) ** i for i, n in enumerate(str(num), 1))
   return '{} !!'.format('Disarium' if total == num else 'Not')


print(disarium_number(89)  , "Disarium !!")
print(disarium_number(518) , "Disarium !!")
print(disarium_number(1024), "Not !!")

File: PHP/7php 1/disariumNumber.php
<?php
/*
|--------------------------------------------------------------------------
| Disarium Number (Special Numbers Series #3)
|--------------------------------------------------------------------------
|
*/

function disariumNumber($n) {
   $str = strval($n);
   $res = 0;
   for ($i = 0; $i < strlen($n); $i++) {
      $res += intval($str[$i]) ** ($i + 1);
   }
   return $res == $n ? "Disarium !!" : "Not !!";
}


function disariumNumber($n) {
   return array_sum(array_map("pow", str_split($n), range(1, strlen($n)))) === $n ?
      "Disarium !!" : "Not !!";
}


function disariumNumber($n) {
   $res = 0;
   foreach (str_split((string)$n) as $k => $v) {
      $res += $v ** ($k + 1); 
   }
   return $res == $n ? "Disarium !!" : "Not !!";
}


function disariumNumber($n) {
   $arr = str_split($n);
   array_walk($arr, function(&$e, $key) {$e = $e ** ($key + 1);});
   return array_sum($arr) == $n ? 'Disarium !!' : 'Not !!';
}


print(disariumNumber(89) . " Disarium !!
");
print(disariumNumber(518) . " Disarium !!
");
print(disariumNumber(1024) . " Not !!
");